arch/arm/core/cortex_m/mpu/nxp_mpu.c

```diff
@@ -328,8 +328,10 @@ static int _mpu_configure_dynamic_mpu_regions(const struct k_mem_partition
 	/* Reset MPU regions inside which dynamic memory regions may
 	 * be programmed.
 	 */
+	arm_core_mpu_disable();
```

Comment:
Could you please, add a short comment, to elaborate why this disable/enable protection is needed here, and not in other invocations of region_init()?
For example:

Re-programming these regions will temporarily leave memory areas outside all MPU regions.
This might trigger memory faults if ISRs occurring during re-programming perform access in
those areas.

Reply:
Thanks! Just added the comment you suggested.